### DISCUSSION TITLE
use different effects on different channels ?

### DISCUSSION THREAD

**Discussion by fjgaston** [garygru/yummyDSP#5] [Q&A] [answered]
Hi,
When using YummyDSP, we can use the 2 channels of our converters but I can't see how to configure some effect nodes only on one of the two channels.

BTW, YummyDSP works fine for me with PCM1802 and PCM5102a converters.

**Comment by garygru** [comment]
Hi, 
yes that's a drawback of the simple design. But you can instantiate two mono yummyDSP classes. One for the left and one for the right channel. Great to hear that it works with your converters, too.

**Comment by fjgaston** [comment]
> But you can instantiate two mono yummyDSP classes

how can I do this ?
Should I have two audiodrivers ?

**Comment by fjgaston** [comment]
Oh I think I can play within this loop in the example
`
    for (int i = 0; i < AudioDriver::BufferSize; i++) {

      for (int ch = 0; ch < channelCount; ch++) {

      sample = i2sCodec.readSample(i, ch);
      
      sample = dsp.process(sample, ch);

      i2sCodec.writeSample(sample, i, ch);
      }`
and have different dsp setup for both channels

**Comment by garygru** [comment, marked as answer]
You need one stereo AudioDriver (i2sCodec) and two mono dsp instances.

**Comment by fjgaston** [comment]
yeah I think I just figured it out.
Thank you for your reactivity
